### ISSUE TITLE
MySQL queries are not always formatted correctly

### ISSUE THREAD

**Issue by isaacabraham** [fsprojects/SQLProvider#347]
For _some_ tables - and I don't know what pattern governs this - queries to MySQL are incorrectly formed and use `""` instead of ``` to surround the table name. Thus, the query is malformed and gets rejected by the database engine.

**Comment by Thorium** [comment]
Do you have tables with whitespace in their name?
Can you give an example of a generated SQL?

**Comment by Thorium** [comment]
For some reason there is this in MySQL code:
```csharp
Table.FullName.Replace("[","`").Replace("]","`")
```
...even though char separator is \":
```fsharp
    let quoteWhiteSpace (str:String) = 
        (if str.Contains(" ") then sprintf "\"%s\"" str else str)
```

so maybe this was just forgotten by some refactoring and 
naïve `Table.FullName.Replace("\"","`").Replace("[","`").Replace("]","`")` would work.

**Comment by pezipink** [comment]
This seems a bit suspicious, MySQL is being used in a bunch of places without this problem, what is causing it to happen in this case?  I think we need to be sure we are not breaking something else with this fix?

**Comment by isaacabraham** [comment]
Sorry for lack of reply. I'm not doing anything special - just doing a select from a table. Even the Individuals property behaves the same way.

There are no tables with spaces in them - just underscores or dashes.

**Comment by pezipink** [comment]
@isaacabraham can you give us an example table name that works and one that doesn't? (I assume this is constant)

Also please let us know what version of the database and drivers you are using?  I am wondering if maybe the metadata is coming back mangled or something is happening between the TP and the actual database call.  Are you looking at the SQL that is hitting the server, what the TP produces, or both?

**Comment by isaacabraham** [comment]
@pezipink How can I see the SQL that the TP produces?


Ooops. Wait. I tell a lie. It looks like it's an table names that have spaces in it that have the problem. All tables with dashes etc. are fine.

**Comment by Thorium** [comment]
```fsharp
 FSharp.Data.Sql.Common.QueryEvents.SqlQueryEvent |> Event.add (printfn "Executing SQL: %s") 
```

As it's only spaces, I'm pretty sure my change will fix this and won't break anything else. I'll merge it now and rollback later if it's not ok.

**Comment by isaacabraham** [comment]
Afraid that this isn't working yet :-( Just tried again. Looks like the query generated is this: -

```sql
SELECT  FROM database.``table`` as `"table"`
```

The correct SQL should be: -

```sql
SELECT * FROM database.`table` as `"table"`
```

Note the missing `*` as well as the double backtick instead of just one.

**Comment by Thorium** [comment]
Ok, was bit more complex than I thought. Fixed in 1.0.37

**Comment by isaacabraham** [comment]
Confirmed now works. Thanks! :-)
